Create map 3

rjwu95 opened the pull request:
백그라운드 색을 0,0,0 으로 설정해놓으면 박스를 생기게 할 수는 있지만 박스를 없앨 수는(충돌하지 않게 할수는) 없어서 초기 배경색을 회색으로 하고 다른 색이 들어왔을 때 배경색이 회색이면 그 색이 배경색이 되는 로직으로 바꿨습니다!

rjwu95 committed:
Create map 3

 - Change logic that compute check initial background color

DoheeKang commented on src/app/Game.js:
```diff
@@ -21,7 +21,7 @@ class Game {
     this.state = GAME_STATE.GAME_READY;
     this.stageNum = 1;
-    this.colorObj = {r: 0, g: 0, b: 0};
+    this.colorObj = {r: 113, g: 107, b: 107};
```
[의견] 하드코딩보다 const에 있는 BACKGROUND_COLOR를 사용하시면 더 좋을 것 같아요

DoheeKang commented on src/app/Game.js:
```diff
@@ -69,14 +69,17 @@ class Game {
-          item.backgroundInitialize = false;
-          item.changeBackground = false;
+        if (
+          (JSON.stringify(colorObj) === JSON.stringify({r: 113, g: 107, b: 107}))
+          || (JSON.stringify(item.color) === JSON.stringify({r: 113, g: 107, b: 107}))
```
이것들도 마찬가지로 상수 처리하면 좋을 것 같습니다.

다만 이렇게 하면 걱정이 단순히 아이템을 먹던 도중에 배경색 (r: 113, g:107, b:107)이 되어버리면 다음 아이템을 먹으면 색이 더해지는게 아니라 해당 아이템 색으로 바껴버리는 현상이 발생할 것 같네요

rjwu95 replied:
앗 그런 이슈가 있네요 그건 어떻게 해결해야 할까요?

rjwu95 replied:
꼼수로 만약에 더해서 초기색이 된다면 rgb중 값 하나를 1만 더하는 것도 방법인것같아요! 더해서 초기색은 될 수 없게!

DoheeKang replied:
@rjwu95 ㅋㅋㅋ음 당장은 시간이 없으니까 섞어서 배경색이 될 수 없게 아이템을 배치하고 추후에 생각해보면 좋을 이슈 같습니다!

DoheeKang commented on src/constant/level.js:
```diff
@@ -21,7 +21,7 @@ export const level = [
     ],
     portal: {x: COLUMNS * 2 - 4, y: 10, color: 'pink'},
-    startPoint: {x:500, y: 0}
+    startPoint: {x:100, y: 200}
```
[의견] 해당 값들은 단순 픽셀값으로 고정시키면 해상도에 따라 위치가 모니터마다 다르게 보이더라구요
map값처럼 COLUMNS, ROWS 값으로 적고 *  BLOCK_SIZE하는게 나을 듯 합니다

rjwu95 committed:
[fix] Apply code review

 - Change starting point using BLOCK_WIDTH and HEIGHT
 - Change order level 2 and 3
 - Change value to constant variables

rjwu95 committed:
Merge branch 'develop' into feature/map-3

rjwu95 commented:
@DoheeKang 바탕색 더해서 회색으로 되는 이슈 제외하고 수정했습니다. 레벨 2,3 도 서로 순서 바꿔서 푸시했어요!
